manager/dispatcher/nodes.go

```diff
@@ -86,8 +89,13 @@ func (s *nodeStore) CheckRateLimit(id string) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 	if existRn, ok := s.nodes[id]; ok {
+		existRn.Attempts++
 		if time.Since(existRn.Registered) < s.rateLimitPeriod {
-			return grpc.Errorf(codes.Unavailable, "node %s attempted registration too recently", id)
+			if existRn.Attempts > rateLimitCount {
+				return grpc.Errorf(codes.Unavailable, "node %s attempted registration too recently", id)
+			}
+		} else {
+			existRn.Attempts = 0
```

Comment:
The field should be time reset, then incremented.

```
if periodHasPassed {
  Attempts = 0
}

Attempts++
if Attempts > periodLimit {
  return Limit
}
```

This is reverse token bucket. Typically, we do:

```
if periodHasPassed {
  Attempts += timeBucketAttempts
  if Attempts > MaxOvershoot {
    Attempts = MaxOvershoot
  }
}

if Attempts < 1 {
  return Limit
}
```

In this approach, we add to the bucket each period, keeping the max capped to allow for overshoot after quiet period.